server/src/grpc/remote_engine_service/mod.rs

```diff
@@ -57,40 +65,130 @@ impl<Q: QueryExecutor + 'static> RemoteEngineServiceImpl<Q> {
         let instant = Instant::now();
         let ctx = self.handler_ctx();
         let (tx, rx) = mpsc::channel(STREAM_QUERY_CHANNEL_LEN);
-        let handle = self.runtimes.read_runtime.spawn(async move {
-            let read_request = request.into_inner();
-            handle_stream_read(ctx, read_request).await
-        });
+
+        let request = request.into_inner();
+        let table_engine::remote::model::ReadRequest {
+            table,
+            read_request,
+        } = request.clone().try_into().box_err().context(ErrWithCause {
+            code: StatusCode::BadRequest,
+            msg: "fail to convert read request",
+        })?;
+
+        let request_key = RequestKey::new(
```

Comment:
I'd like to make this feature optional because it introduces some overhead, that is to say, if this feature is disabled, the procedure should be kept as before.